### ISSUE TITLE
Dramatically different model parameter estimates and tree length between 2.2.0 and 2.1.4

### ISSUE THREAD

**Issue by rneher** [iqtree/iqtree2#53]
We have been running IQ tree on simulated data along an HIV phylogeny and obtain dramatically different results when using version 2.2.0 vs 2.1.4. Total tree length differs by a factor of 3 and the estimated rate categories and distribution are very different. 

The alignment used was:

[aln.fasta.gz](https://github.com/iqtree/iqtree2/files/7840179/aln.fasta.gz)

IQ-tree was run as 

`iqtree2 -ninit 2 -n 2 -me 0.05 -nt 4 -s IQtree_2.2.0/aln.fasta -m GTR+F+R10  -czb`

For version 2.2.0, we obtain a tree-length of 96 and a wide rate distribution peaked at 0. 

```
Rate parameters:  A-C: 1.71553  A-G: 56.49825  A-T: 0.40073  C-G: 0.74865  C-T: 100.00000  G-T: 1.00000
Base frequencies:  A: 0.385  C: 0.169  G: 0.234  T: 0.212
Site proportion and rates:  (0.263,0.050) (0.139,0.145) (0.130,0.239) (0.109,0.396) (0.119,0.629) (0.081,0.933) (0.050,1.492) (0.045,2.557) (0.020,5.333) (0.044,10.219)

[...]

BEST SCORE FOUND : -404620.664
Collapsing near-zero internal branches... 41 collapsed
Total tree length: 95.702
```

For version 2.1.4, total tree length is 27 and the rate distribution is peaked around 1. 
```
Optimal log-likelihood: -403882.383
Rate parameters:  A-C: 2.14594  A-G: 58.12361  A-T: 0.56247  C-G: 0.93941  C-T: 100.00000  G-T: 1.00000
Base frequencies:  A: 0.385  C: 0.169  G: 0.234  T: 0.212
Site proportion and rates:  (0.098,0.121) (0.185,0.177) (0.010,0.308) (0.156,0.480) (0.117,0.698) (0.115,1.068) (0.149,1.657) (0.117,2.134) (0.043,3.103) (0.009,4.676)

[...]

BEST SCORE FOUND : -403378.895
Collapsing near-zero internal branches... 43 collapsed
Total tree length: 33.723
```

**Comment by rneher** [comment]
Looking at the changelog of previous releases, my hunch is that this has to do with the BGFS optimization. Also, the log states that parameters estimates are on the boundary of the domain.

**Comment by bqminh** [comment]
Thanks, this is an important bug, which is now fixed with commit 435596cac30a820
